### ISSUE TITLE
GRPO Trainer/Config Incomplete metrics logging to wandb

### ISSUE THREAD

**Issue by SpaceHunterInf** [huggingface/trl#3078]
### Reproduction

Greetings,

In version `0.15.2`, metrics such as kl and loss are no longer uploaded to wandb for GRPO Trainer. May I ask if this is expected? If yes, what's the best practice for me to enable these logging manually again?

Currently I am doing the logging by simply passing the argument `report_to='wandb'` to the GRPO Config.

Many thanks.

### System Info

- Platform: Linux-6.2.0-39-generic-x86_64-with-glibc2.35
- Python version: 3.12.9
- PyTorch version: 2.5.1
- CUDA device(s): Quadro RTX 8000
- Transformers version: 4.49.0
- Accelerate version: 1.4.0
- Accelerate config: not found
- Datasets version: 3.3.2
- HF Hub version: 0.29.2
- TRL version: 0.15.2
- bitsandbytes version: 0.45.3
- DeepSpeed version: not installed
- Diffusers version: 0.32.2
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: 1.65.4
- PEFT version: 0.14.0

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by tchang1997** [comment]
What's your exact set of trainer args/training script? I noticed that if `self.beta == 0.0`, KL logging is skipped altogether, though that change may have been after 0.15.2.

**Comment by SpaceHunterInf** [comment]
Thanks for helping me.

I am actually using unsloth + trl GPRO. I can see the cmd output doing well on wandb log, but not on the wandb workspace. The log itself contains all the things I need.

```{'loss': 0.0133, 'grad_norm': 2.667192220687866, 'learning_rate': 2.6140692393428204e-10, 'rewards/correctness_reward_func': 0.25, 'rewards/confidence_reward_func': 0.0, 'rewards/int_reward_func': 0.5, 'rewards/soft_format_reward_func': 0.5, 'reward': 1.25, 'reward_std': 1.3363062143325806, 'completion_length': 32.0, 'kl': 0.3319159746170044, 'epoch': 0.92}```

TLDR, I attached my training config and scripts below.

**Config**
```
cfg = TrainingConfig(
    dataset_name="ecqa",
    
    # Model settings
    model_name="Qwen/Qwen2.5-1.5B-Instruct",
    max_seq_length=1024,
    lora_rank=128,
    load_in_4bit=True,
    fast_inference=True,
    gpu_memory_utilization=0.8,
    target_modules=[
        "q_proj", "k_proj", "v_proj", "o_proj",
        "gate_proj", "up_proj", "down_proj",
    ],
    
    # Training args
    use_vllm=True,
    learning_rate=5e-6,
    adam_beta1=0.9,
    adam_beta2=0.99,
    weight_decay=0.1,
    warmup_ratio=0.1,
    lr_scheduler_type="cosine",
    optim="adamw_8bit",
    logging_steps=1,
    per_device_train_batch_size=8,
    gradient_accumulation_steps=1,
    num_generations=8,
    max_prompt_length=256,
    max_completion_length=32,
    max_steps=7000,
    save_steps=1400,
    max_grad_norm=0.1,
    report_to="wandb",
    output_dir="outputs",
    wandb_name="ecqa_qwen_hpc",
    
    # Reward functions
    reward_funcs=[
        "correctness_reward_func",
        "confidence_reward_func",
        "int_reward_func",
        "soft_format_reward_func"
    ]
)
```


**Training Script**
```
import argparse
import json
import os, sys
from pathlib import Path
import torch
from unsloth import FastLanguageModel, is_bfloat16_supported
from trl import GRPOConfig, GRPOTrainer

from utils.data_utils import get_dataset
from utils.training_utils import (
    correctness_reward_func, confidence_reward_func, 
    int_reward_func, soft_format_reward_func, load_config, save_config
)

def get_reward_functions(reward_func_names):
    """Map reward function names to actual functions"""
    reward_funcs_map = {
        "correctness_reward_func": correctness_reward_func,
        "confidence_reward_func": confidence_reward_func,
        "int_reward_func": int_reward_func,
        "soft_format_reward_func": soft_format_reward_func
    }
    
    return [reward_funcs_map[name] for name in reward_func_names]

def train(cfg):
    """Main training function"""
    # Load dataset
    train_dataset, dev_dataset, test_dataset = get_dataset(cfg.dataset_name)
    
    # Initialize model
    model, tokenizer = FastLanguageModel.from_pretrained(
        model_name=cfg.model_name,
        max_seq_length=cfg.max_seq_length,
        load_in_4bit=cfg.load_in_4bit,
        fast_inference=cfg.fast_inference,
        max_lora_rank=cfg.lora_rank,
        gpu_memory_utilization=cfg.gpu_memory_utilization,
    )
    
    model = FastLanguageModel.get_peft_model(
        model,
        r=cfg.lora_rank,
        target_modules=cfg.target_modules,
        lora_alpha=cfg.lora_rank,
        use_gradient_checkpointing="unsloth",
        random_state=3407,
    )
    
    # Configure training arguments
    training_args = GRPOConfig(
        use_vllm=cfg.use_vllm,
        learning_rate=cfg.learning_rate,
        adam_beta1=cfg.adam_beta1,
        adam_beta2=cfg.adam_beta2,
        weight_decay=cfg.weight_decay,
        warmup_ratio=cfg.warmup_ratio,
        lr_scheduler_type=cfg.lr_scheduler_type,
        optim=cfg.optim,
        logging_steps=cfg.logging_steps,
        bf16=is_bfloat16_supported(),
        fp16=not is_bfloat16_supported(),
        per_device_train_batch_size=cfg.per_device_train_batch_size,
        gradient_accumulation_steps=cfg.gradient_accumulation_steps,
        num_generations=cfg.num_generations,
        max_prompt_length=cfg.max_prompt_length,
        max_completion_length=cfg.max_completion_length,
        max_steps=cfg.max_steps,
        save_steps=cfg.save_steps,
        max_grad_norm=cfg.max_grad_norm,
        report_to=cfg.report_to,
        output_dir=cfg.output_dir,
    )
    
    if cfg.num_train_epochs is not None:
        training_args.num_train_epochs = cfg.num_train_epochs
    
    # Set up reward functions
    reward_funcs = get_reward_functions(cfg.reward_funcs)
    
    # Initialize trainer
    trainer = GRPOTrainer(
        model=model,
        processing_class=tokenizer,
        reward_funcs=reward_funcs,
        args=training_args,
        train_dataset=train_dataset,
        eval_dataset=dev_dataset,
    )
    
    # Start training
    trainer.train()

def main():
    
    parser = argparse.ArgumentParser(description="RL training script")
    parser.add_argument("--config", type=str, default=None, help="Path to config file")
    args = parser.parse_args()
    
    # Load configuration
    
    cfg = load_config(args.config)
    
    if cfg.report_to == 'wandb':
        import wandb
        os.environ['WANDB_API_KEY'] = 'd2897fab8b1ef5d7affd6fc9f1295f37d78c891f'
        wandb.init(project="unsloth", config=cfg.__dict__, name=cfg.wandb_name)
    # Save configuration
    save_config(cfg, cfg.output_dir)
    
    # Start training
    train(cfg)
    if cfg.report_to == 'wandb':
        wandb.finish()
```

**Comment by tchang1997** [comment]
Huh, that's odd. Just to confirm, you're seeing all the other metrics, just not KL and loss? Just to make sure this isn't an unsloth thing, maybe try training w/ a tiny mockup dataset w/o unsloth? You might also try explicitly setting `beta` to something non-zero in the `GRPOConfig.`

FWIW, I haven't needed to explicitly `wandb.init` in my main script — I just let the trainer take care of that and set `WANDB_PROJECT="unsloth"` in the env, since I had some issues with duplicated runs/metrics not logging where I expected on wandb. But if logging was working before this is unlikely to be the issue.

**Comment by SpaceHunterInf** [comment]
Here is a screen shot of my wandb workspace, one suspicious thing that I noticed is on the top left it says (6 of 13) but on the top right it says (1-6 of 6).

<img width="1446" alt="Image" src="https://github.com/user-attachments/assets/85013b62-0e8c-41cf-973a-bf041b2738c9" />

I will try to see if the beta thing works. I use wandb.init because I want to associate each run on wandb with my model settings as their wandb name.

Thanks

**Comment by SpaceHunterInf** [comment]
... Alright, I know the reason why.. I am an absolute idiot.
I didn't realize there was something in the regex filter on wandb. Everything is indeed there.
